addon/pods/components/frost-tab/component.js

```diff
@@ -1,21 +1,59 @@
 import Ember from 'ember'
 import layout from './template'
-import FrostTabs from 'ember-frost-tabs/pods/components/frost-tabs/component'
+import PropTypesMixin, { PropTypes } from 'ember-prop-types'
+
+const {
+  Component,
+  computed
+} = Ember
+
+export default Component.extend(PropTypesMixin, {
+  // == Component properties ==================================================
 
-export default Ember.Component.extend({
   layout: layout,
-  classNames: ['content'],
-  classNameBindings: ['isSelected::hidden', 'tabClassNames'],
+  classNames: ['frost-tab'],
 
-  frostTabs: null,
+  // == State properties ======================================================
+
+  propTypes: {
+    id: PropTypes.string.isRequired,
+    text: PropTypes.string.isRequired,
+    content: PropTypes.oneOfType([
+      PropTypes.EmberObject,
+      PropTypes.object
+    ]).isRequired,
+    disabled: PropTypes.bool,
+    // Set by the parent component
+    hook: PropTypes.string,
+    parentHook: PropTypes.string,
+    targetOutlet: PropTypes.string.isRequired,
+    selectedTab: PropTypes.string.isRequired,
+    onChange: PropTypes.func.isRequired
+  },
 
-  init () {
-    this._super(...arguments)
-    this.set('frostTabs', this.nearestOfType(FrostTabs))
-    this.get('frostTabs').register(this)
+  getDefaultProps () {
+    return {
+      disabled: false
+    }
   },
 
-  isSelected: Ember.computed('frostTabs.selection', function () {
-    return this.get('id') === this.get('frostTabs.selection')
-  })
+  // == Computed properties ===================================================
+
+  isSelected: computed('id', 'selectedTab', function () {
+    return this.id === this.selectedTab && !this.disabled
+  }),
+
+  hook: computed('parentHook', 'id', function () {
+    return `${this.parentHook}-${this.id}`
+  }),
+
+  // == Actions ===============================================================
+
+  actions: {
+    change () {
```

Comment:
Ah, this is what I was looking for - as soon as you do that set you'll be disconnecting the observer chain, so we can no longer programatically change the selectedTab.  In addition, this would result in state being held in both the controller (via the onChange and initial? selectedTab) and inside the component, which is an anti-pattern.

I know it seems like boilerplate to require the consumer to wire up both the downstream data and upstream event, but that's best practice state management, so we should stick with that approach.

If the consumer wants to short circuit this, they can use the `mut` helper with `onChange`